[W5.7][W16-1]Javen Seow Yu Bin

javenseow opened the pull request:
Added contain command

javenseow committed:
ContainCommand.java: Added contain command
ContainCommandTest.java: Added tests for contain command
Parser.java: Updated parser with new command
HelpCommand.java: Updated help page with contain command

javenseow committed:
updated output file

kengwoon commented:
Can add more test cases, but other than that, all is good.

GilgameshTC commented:
It would be better if the changes were done incrementally, but great job none-the-less!

ericyjw commented:
Good job! However, you could have added comments before the methods!

kengwoon commented on src/seedu/addressbook/commands/ContainCommand.java:
```diff
@@ -0,0 +1,36 @@
+
+    @Override
+    public CommandResult execute() {
+        boolean personFound = checkDuplicate(name);
```
can name variable as isPersonFound

bqnguyen94 commented on src/seedu/addressbook/parser/Parser.java:
```diff
@@ -97,12 +87,25 @@ public Command parseCommand(String userInput) {
         }
     }
 
+    private Command prepareContain(String args) {
```
Would be nice if this function has explanatory comments as well, see the `prepareAdd` function below :)

bqnguyen94 commented on src/seedu/addressbook/parser/Parser.java:
```diff
@@ -11,17 +11,7 @@
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
These style changes are not really relevant to the purpose of this PR which is adding the `contain` command.

bqnguyen94 commented on test/expected.txt:
```diff
@@ -29,6 +29,7 @@
 || Example: help
 || exit: Exits the program.
 || Example: exit
+|| contain: Searches the address book with given name
```
How about an usage example as well? :)

javenseow replied:
Added an example!

bqnguyen94 commented on test/java/seedu/addressbook/commands/ContainCommandTest.java:
```diff
@@ -0,0 +1,71 @@
+    public void execute_validAddressBook_personNotContained() throws IllegalValueException {
+        assertContainSuccessful("Lili Pan", addressbook, false);
+    }
+    private void assertContainSuccessful(String name, AddressBook addressbook, boolean contained) throws IllegalValueException {
```
Nitpick: leave a line between functions to separate them.

bqnguyen94 commented on test/java/seedu/addressbook/parser/ParserTest.java:
```diff
@@ -12,17 +12,7 @@
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
These changes are also not really relevant?

bqnguyen94 reviewed (changes requested):
Almost there 👍

bqnguyen94 reviewed (changes requested):
Pls update the user guide as well

javenseow committed:
ContainCommand.java: changed personFound variable to isPersonFound
Parser.java: added comments for prepareContain and changed the import format
expected.txt: Added usage example
ContainCommandText.java: added a line in between functions
ParserTest.java: changed the import format
UserGuide.adoc: updated UserGuide with contain command

javenseow committed:
ParserTest.java: added a line to separate import statements

bqnguyen94 commented on test/java/seedu/addressbook/parser/ParserTest.java:
```diff
@@ -23,6 +24,7 @@
 import seedu.addressbook.commands.ListCommand;
 import seedu.addressbook.commands.ViewAllCommand;
 import seedu.addressbook.commands.ViewCommand;
+
```
This is not needed. Normally we only separate imports that are from different parties.

bqnguyen94 reviewed (approved):
Good job 👍 you can close the PR now.
